tests/testthat/_snaps/variables.md

```diff
@@ -0,0 +1,86 @@
+# variable() errors informatively
+
+    Code
+      variable(upper = NA)
+    Error <simpleError>
+      lower and upper must be numeric
+      lower has class: numeric
+      lower has length: 1
+      upper has class: logical
+      upper has length: 1
+
+---
+
+    Code
+      variable(upper = head)
+    Error <simpleError>
+      lower and upper must be numeric
+      lower has class: numeric
+      lower has length: 1
+      upper has class: function
+      upper has length: 1
+
+---
+
+    Code
+      variable(lower = 0:2, upper = 1:2)
+    Error <simpleError>
+      incompatible dimensions: 3x1, 2x1
+
+---
+
+    Code
+      variable(lower = 1, upper = 1)
+    Error <simpleError>
+      upper bounds must be greater than lower bounds
+      lower is: 1
+      upper is: 1
+
+# constrained variable constructors error informatively
+
+    Code
+      cholesky_variable(dim = 2:3)
+    Error <simpleError>
+      cholesky variables must be square
+      However its dimension is: "2x3"
+
+---
+
+    Code
+      cholesky_variable(dim = rep(2, 3))
+    Error <simpleError>
+      dim can either be a scalar or a vector of length 2
+      However its dimension is: 2, 2, and 2
```

Comment:
shouldn't say 'its dimension', because the 'it' here _is_ the dimension

Reply:
I'm not sure how to rephrase this - would something like this work in this instance?

```
dim can either be a scalar or a vector of length 2
but dim has length of 3: 2, 2, 2
```

Reply:
yeah, that's perfect